### ISSUE TITLE
Request to compare RIFE v4.15 and v4.17 with GIF files of the InterpAny-Clearer demo

### ISSUE THREAD

**Issue by AIVFI** [hzwer/Practical-RIFE#69]
I would like to thank you very much for continually adding new, practical RIFE models. For this last one I am particularly grateful to you. Firstly because you are open to the work of others that can improve the already awesome RIFE. Secondly, because I was very interested in adding just this loss function to RIFE, which I requested from the developer InterpAny-Clearer here: [Request for [D,R] RIFE trained on Style loss, also called Gram matrix loss (the best perceptual loss function)](https://github.com/zzh-tech/InterpAny-Clearer/issues/12)

As you have applied gram matrix loss to RIFE v4.17 training, I would like to ask you to conduct x128 frame interpolation tests of RIFE v4.15 and v4.17 for the files I0_0.png and I0_1.png and also I1_0.png and I1_1.png available here: https://github.com/zzh-tech/InterpAny-Clearer/tree/main/demo and posting their results as GIF files in the Practical-RIFE repository or here. There will be a total of 4 GIF files.

I think this test will be of interest to any Practical-RIFE enthusiast and will answer an important question: can gram matrix loss really improve the performance of practical RIFE models? And, in particular, whether it eliminates or at least mitigates to some extent the undesirable distortions occurring with VGG loss, which can be seen particularly clearly in the lower example for the column: [D,R] RIFE-vgg (Ours) in the table available at this link: https://github.com/zzh-tech/InterpAny-Clearer?tab=readme-ov-file#time-indexing-vs-distance-indexing

This test will not only compare the differences in the loss function for version 4.15 and version 4.17 of RIFE, but will also give an answer to the question of whether the application of the InterpAny-Clearer enhancement will further improve the clarity of frame interpolation for practical RIFE models, just as it improves it for the base RIFE model, as shown in the table to which I have provided a link above.

We will have a total comparison of 5 different versions of RIFE - 3 already available:

[T] RIFE - RIFE base model
[D,R] RIFE - RIFE base model with full InterpAny-Clearer upgrade
[D,R] RIFE-vgg - RIFE base model with VGG loss and full InterpAny-Clearer upgrade

and 2 versions of the practical RIFE models, which I ask you to compare here:

RIFE v4.15 with standard perceptual loss
RIFE v4.17 with gram matrix loss

This comparison will hopefully give evidence to justify the use of gram matrix loss in all future RIFE practical models. Also, I hope it will demonstrate the need for the use of InterpAny-Clearer in the next RIFE v4.18 model.

I would also like to use these GIF files of 5 different versions of RIFE (of course with links to the sources of the comparisons) in the introduction to my rankings [Video Frame Interpolation Rankings and Video Deblurring Rankings](https://github.com/AIVFI/Video-Frame-Interpolation-Rankings-and-Video-Deblurring-Rankings), where I would like to encourage other researchers to use the best and above all proven solutions for practical applications when developing new models.

I hope that the next GIF file will already be for RIFE v4.18 with gram matrix loss and the InterpAny-Clearer enhancement. The clarity of the frame interpolation is more important than the super resolution, especially when new monitors already reach 500Hz and the original frames can be seen on the screen very briefly and most of the time the interpolated frames are visible.

**Comment by hzwer** [comment]
I feel a little confused. I found that gram matrix loss mainly affects the performance of anime scenes. I don't know if most people will like it.

a hard case example:
![image](https://github.com/hzwer/Practical-RIFE/assets/10103856/e906597f-6be4-4c71-844a-e6b6a375fe4e)

Also, I'm concerned that comparing [D, R] RIFE to this project may be misleading:
In order to improve performance, Practical RIFE invests in more training data, computing power, data augmentation and lots of tricks. InterpAny-Clearer doesn't seem to be published currently and I'm worried about putting pressure on the authors.
I'm interested in trying to incorporate InterpAny-Clearer's technology, but it will take many experiments.

**Comment by AIVFI** [comment]
> I feel a little confused. I found that gram matrix loss mainly affects the performance of anime scenes. I don't know if most people will like it.
> 
> some hard examples: ![image](https://private-user-images.githubusercontent.com/10103856/333865855-e906597f-6be4-4c71-844a-e6b6a375fe4e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY3NjcyNTcsIm5iZiI6MTcxNjc2Njk1NywicGF0aCI6Ii8xMDEwMzg1Ni8zMzM4NjU4NTUtZTkwNjU5N2YtNmJlNC00YzcxLTg0NGEtZTZiNmEzNzVmZTRlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI2VDIzNDIzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliY2NkODVmY2Y1ZmUxZWZlZTdjN2UxM2IyNDI4MDJmZDg3NjQ3MjEyM2E1NzIxNmI2YWIyY2IwMjA3ODNlMjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UHWT4g08JZS0hY40kngDGJT2lXMpGhikkasb8oIAgww) ![504330f9576f9ce70fd01b2080798998](https://private-user-images.githubusercontent.com/10103856/333866206-aea8eae0-8924-46aa-acbb-64c82d802279.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY3NjcyNTcsIm5iZiI6MTcxNjc2Njk1NywicGF0aCI6Ii8xMDEwMzg1Ni8zMzM4NjYyMDYtYWVhOGVhZTAtODkyNC00NmFhLWFjYmItNjRjODJkODAyMjc5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI2VDIzNDIzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhZWEwZjFiNTM1ZTM1NGQzMzY0OGQ5N2M0MzA1YzE5NzM1MDJiMDdlOWU3Njk0Yjg5NmFmZmQ3OTMxMmU1ZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.INNmIUJ963IuvfJggzQM6j6aT_fOtTfK24490OHruPU)
> 
> Also, I'm concerned that the demo of comparing [D, R] RIFE to this project may be misleading: In order to improve performance, Practical RIFE invests in more training data, computing power, data augmentation and lots of tricks. InterpAny-Clearer doesn't seem to be published currently and I'm worried about putting pressure on the authors. I'm interested in trying to incorporate InterpAny-Clearer's technology, but it will take many experiments.

**Comment by AIVFI** [comment]
> I'm interested in trying to incorporate InterpAny-Clearer's technology, but it will take many experiments.

Great news! Thank you! This is the most important thing for me as a frame interpolation enthusiast. I understand that it won't be easy or fast, but many people will be very grateful to you for this.

> InterpAny-Clearer doesn't seem to be published currently and I'm worried about putting pressure on the authors.

Unfortunately, I am not a programmer and was not aware that the code for [InterpAny-Clearer](https://github.com/zzh-tech/InterpAny-Clearer) has not been published in full. So far, [zzh-tech](https://github.com/zzh-tech) has been eager to share his achievements, including his revolutionary [real-world dataset RBI](https://github.com/zzh-tech/BiT?tab=readme-ov-file#preparation), which allows to train frame interpolation models to remove the worst artefacts caused by motion blur, which is larger the more dynamic the movement of a person or object between frames. These artefacts cannot be eliminated by training with either the X4K1000FPS dataset or the SNU-FILM dataset, because this is impossible to achieve with a dataset made with a single camera. So since he has shared such a treasure as the [RBI dataset](https://github.com/zzh-tech/BiT?tab=readme-ov-file#preparation), I think he will also share the code and help with the [InterpAny-Clearer](https://github.com/zzh-tech/InterpAny-Clearer) implementation. Especially since he responded positively to my request and was open to collaboration: https://github.com/zzh-tech/InterpAny-Clearer/issues/12#issuecomment-2005793456

**Comment by AIVFI** [comment]
> I feel a little confused. I found that gram matrix loss mainly affects the performance of anime scenes. I don't know if most people will like it.

Yes, gram matrix loss affects the performance of anime scenes, but it also improves live action videos, as can be seen in [Fig. 1](https://www.ecva.net/papers/eccv_2022/papers_ECCV/papers/136670244-supp.pdf) in Supplementary Material.

However, the most interesting thing is this visual comparison of the three loss functions. In my opinion, Style loss clearly gives the best result perceptually:

![FILM - Loss Functions Ablation](https://film-net.github.io/static/images/loss_ablation.png)
Source: FILM - Loss Functions Ablation https://film-net.github.io/

The first time Style loss to train the video frame interpolation model was used by **Google Research** when training their FILM-𝓛<sub>*S*</sub> model, link: [![ECCV](https://img.shields.io/badge/2022-ECCV-67cd84)](https://www.ecva.net/papers/eccv_2022/papers_ECCV/html/4614_ECCV_2022_paper.php) The second time this loss was used by **Disney Research** when training their UGFI 𝓛<sub>*S*</sub> model, link: [![CVPR](https://img.shields.io/badge/2023-CVPR-1e407f)](https://openaccess.thecvf.com/content/CVPR2023/html/Plack_Frame_Interpolation_Transformer_and_Uncertainty_Guidance_CVPR_2023_paper.html) 

Both models achieved some of the best and maybe even the best LPIPS results (because we do not have a direct comparison with the other two models of the top four, and the results are very close to each other):

**Vimeo-90K triplet: LPIPS<=0.017 [excluding LPIPS(SqueezeNet) results]**
| Rank | &nbsp;&nbsp;&nbsp;&nbsp;Model&nbsp;&nbsp;&nbsp;&nbsp; | &nbsp;&nbsp;&nbsp;LPIPS&nbsp;↓&nbsp;&nbsp;&nbsp;<br />{Input&nbsp;fr.} | Training<br />dataset | Official<br />&nbsp;&nbsp;repository&nbsp;&nbsp; | Practical<br />model | VapourSynth |
|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
| 1 | EAFI-𝓛<sub>*ecp*</sub><br />[![arXiv](https://img.shields.io/badge/2022-arXiv-b31b1b)](https://arxiv.org/abs/2207.12305) |  **0.012** {2}<br />[![arXiv](https://img.shields.io/badge/2022-arXiv-b31b1b)](https://arxiv.org/abs/2207.12305) | Vimeo-90K triplet | - | EAFI-𝓛<sub>*ecp*</sub> | - |
| 2 | UGFI 𝓛<sub>*S*</sub><br />[![CVPR](https://img.shields.io/badge/2023-CVPR-1e407f)](https://openaccess.thecvf.com/content/CVPR2023/html/Plack_Frame_Interpolation_Transformer_and_Uncertainty_Guidance_CVPR_2023_paper.html) | **0.0126** {2}<br />[![CVPR](https://img.shields.io/badge/2023-CVPR-1e407f)](https://openaccess.thecvf.com/content/CVPR2023/html/Plack_Frame_Interpolation_Transformer_and_Uncertainty_Guidance_CVPR_2023_paper.html) | Vimeo-90K triplet | - | UGFI 𝓛<sub>*S*</sub> | - |
| 3 | SoftSplat - 𝓛<sub>*F*</sub><br />[![CVPR](https://img.shields.io/badge/2020-CVPR-1e407f)](https://openaccess.thecvf.com/content_CVPR_2020/html/Niklaus_Softmax_Splatting_for_Video_Frame_Interpolation_CVPR_2020_paper.html) |  **0.013** {2}<br />[![CVPR](https://img.shields.io/badge/2020-CVPR-1e407f)](https://openaccess.thecvf.com/content_CVPR_2020/html/Niklaus_Softmax_Splatting_for_Video_Frame_Interpolation_CVPR_2020_paper.html) | Vimeo-90K triplet | [![GitHub Stars](https://img.shields.io/github/stars/sniklaus/softmax-splatting?logo=GitHub&label=Stars)](https://github.com/sniklaus/softmax-splatting) | SoftSplat - 𝓛<sub>*F*</sub> | - |
| 4 | FILM-𝓛<sub>*S*</sub><br />[![ECCV](https://img.shields.io/badge/2022-ECCV-67cd84)](https://www.ecva.net/papers/eccv_2022/papers_ECCV/html/4614_ECCV_2022_paper.php) | **0.0132** {2}<br />[![CVPR](https://img.shields.io/badge/2023-CVPR-1e407f)](https://openaccess.thecvf.com/content/CVPR2023/html/Plack_Frame_Interpolation_Transformer_and_Uncertainty_Guidance_CVPR_2023_paper.html) | Vimeo-90K triplet | [![GitHub Stars](https://img.shields.io/github/stars/google-research/frame-interpolation?logo=GitHub&label=Stars)](https://github.com/google-research/frame-interpolation) | [FILM-𝓛<sub>*S*</sub>](https://github.com/google-research/frame-interpolation#pre-trained-models) | - |
| 5 | EDSC_s-𝓛<sub>*F*</sub><br />[![TPAMI](https://img.shields.io/badge/2021-TPAMI-fefd02)](https://ieeexplore.ieee.org/document/9501506) |  **0.016** {2}<br />[![arXiv](https://img.shields.io/badge/2020-arXiv-b31b1b)](https://arxiv.org/abs/2006.08070) | Vimeo-90K triplet | [![GitHub Stars](https://img.shields.io/github/stars/Xianhang/EDSC-pytorch?logo=GitHub&label=Stars)](https://github.com/Xianhang/EDSC-pytorch) | [EDSC_s-𝓛<sub>*F*</sub>](https://github.com/Xianhang/EDSC-pytorch#pre-trained-models) | - |
| 6 | CtxSyn - 𝓛<sub>*F*</sub><br />[![CVPR](https://img.shields.io/badge/2018-CVPR-1e407f)](https://openaccess.thecvf.com/content_cvpr_2018/html/Niklaus_Context-Aware_Synthesis_for_CVPR_2018_paper.html) | **0.017** {2}<br />[![CVPR](https://img.shields.io/badge/2020-CVPR-1e407f)](https://openaccess.thecvf.com/content_CVPR_2020/html/Niklaus_Softmax_Splatting_for_Video_Frame_Interpolation_CVPR_2020_paper.html) | proprietary | - | CtxSyn - 𝓛<sub>*F*</sub> | - |
| 7 | PerVFI<br />[![CVPR](https://img.shields.io/badge/2024-CVPR-1e407f "The link to the conference paper will be available in June. CVPR 2024: June 17th – 21st")](https://cvpr.thecvf.com/Conferences/2024/AcceptedPapers) | **0.018** {2}<br />[![arXiv](https://img.shields.io/badge/2024-arXiv-b31b1b)](https://arxiv.org/abs/2404.06692) | Vimeo-90K triplet | [![GitHub Stars](https://img.shields.io/github/stars/mulns/PerVFI?logo=GitHub&label=Stars)](https://github.com/mulns/PerVFI) | PerVFI | - |

**Therefore, please continue to use this loss function in future practical RIFE models.**

**Comment by AIVFI** [comment]
> Also, I'm concerned that the demo of comparing [D, R] RIFE to this project may be misleading: In order to improve performance, Practical RIFE invests in more training data, computing power, data augmentation and lots of tricks. 

Please, incorporate InterpAny-Clearer's technology in Practical-RIFE. I changed the title a minute ago, as this is the most important thing for me. I realise how different Practical-RIFE is from other models that, for publication at conferences, are generally only trained on one database and how this affects their practical application.

Due to your commitment, I dream that maybe one day you will undertake to train a practical model of Joint Video Deblurring and Frame Interpolation.
